File: Apps/BankCore/backend/src/main/java/org/d3javu/backend/model/business/payment/Payment.java
package org.d3javu.backend.model.business.payment;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.d3javu.backend.model.base.personalaccount.PersonalAccount;
import org.d3javu.backend.model.business.invoice.Invoice;
import org.d3javu.backend.model.business.paymentaccount.PaymentAccount;
import org.d3javu.backend.model.transaction.Transaction;

import java.util.UUID;

@Entity
@Table(name = "payment")
@Getter
@Setter
@NoArgsConstructor
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne
    @JoinColumn(name = "payment_account_id")
    private PaymentAccount paymentAccount;

    @OneToOne
    @JoinColumn(name = "personal_account_id")
    private PersonalAccount personalAccount;

    @OneToOne
    @JoinColumn(name = "transaction_id")
    private Transaction transactionId;

    @OneToOne
    @JoinColumn(name = "invoice_id")
    private Invoice invoice;

}

File: Apps/BankCore/backend/src/main/scala/domain/business/Payment.scala
package org.d3javu
package domain.business

import java.util.UUID

import org.d3javu.domain.base.PersonalAccount
import org.d3javu.domain.business.Payment.Id
import org.d3javu.domain.transaction.Transaction

import io.estatico.newtype.macros.newtype

case class Payment(
    id: Id,
    paymentAccountId: PaymentAccount.Id,
    personalAccountId: PersonalAccount.AccountId,
    transactionId: Transaction.Id,
    invoiceId: Invoice.Id,
)

object Payment {

  @newtype
  final case class Id(asUUID: UUID)

}
